### DISCUSSION TITLE
Is ApplicationSet applicable to promotion, e.g, staging to prod?

### DISCUSSION THREAD

**Discussion by joebowbeer** [argoproj-labs/applicationset#240] [Q&A]
ApplicationSet is addressing some multi-cluster use cases, which makes me wonder about situations where the clusters represent stages, such as staging and prod, and whether ApplicationSet is intended to be used as a mechanism for managing the promotion of apps from one stage to the next.

After a closer look at the docs, I think the answer is no. Correct?

In other words, it looks like ApplicationSet is designed for multiplexing one-to-many but is not designed for promotion.

Assuming I'm correct, I think it would be helpful to make this clear in the docs. If I'm not correct, I'd love to see samples of promotion.

**Comment by rumstead** [comment]
I was planning to use it for promotion across our many preproduction environments and many production environments. I haven't come up with a plan yet but some ideas.

1. Use the [config.json approach here](https://argocd-applicationset.readthedocs.io/en/stable/Generators/#git-generator-files)
    * Little nasty because we will need a config per environment 
2. Some matrix of generators and filters 

I would love it if others have a solution/recommendations. Or tell me I am crazy and not to do it. 

EDIT: Attaching similar-esk issues, #221 and #61

**Comment by ghost** [comment]
I believe this use case is being handled here?
https://github.com/argoproj-labs/applicationset/issues/221
